clinica/iotools/converters/oasis3_to_bids/oasis3_to_bids.py

```diff
@@ -0,0 +1,165 @@
+            import shutil
+
+            def copy_file(source, target):
+                try:
+                    new_filename = re.sub("OAS3", "OASIS3", path.basename(source))
+                    new_filename = re.sub("_echo-", "_run-", new_filename)
+                    if re.search(".*_pet.(nii.gz|json)", new_filename) is not None:
+                        new_filename = re.sub("task-rest_", "", new_filename)
+                        new_filename = re.sub("acq", "trc", new_filename)
+                        target = path.join(path.dirname(target), "pet")
+                    elif re.search(".*_(minIP|swi|dwi).(nii.gz|json)", new_filename) is not None:
+                        target = path.join(path.dirname(target), "swi")
+                    if not os.path.isdir(target):
+                        os.mkdir(path.join(target))
+                    shutil.copy(source, target)
+                    os.rename(
+                        path.join(target, path.basename(source)),
+                        path.join(target, new_filename)
+                    )
+                except IOError as e:
+                    print("Unable to copy file. %s" % e)
+
+            subj_id = os.path.basename(subj_folder)
+            print("Converting ", subj_id)
+            old_sess_fold = subj_id.split("_")
+            numerical_id = re.sub(r'^OAS3', '', old_sess_fold[0])
+            bids_id = "sub-OASIS3" + str(numerical_id)
+            bids_subj_folder = path.join(dest_dir, bids_id)
+            if not os.path.isdir(bids_subj_folder):
+                os.mkdir(bids_subj_folder)
+
+            session_name = old_sess_fold[2]
+            session_folder = path.join(bids_subj_folder, "ses-" + session_name)
+            if not os.path.isdir(session_folder):
+                os.mkdir(path.join(session_folder))
+
+            # In order do convert the Analyze format to Nifti the path to the .img file is required
+            nifti_folders = glob(
+                path.join(subj_folder, "**/*.nii.gz"), recursive=True
+            )
+            #TODO: if not nifit ( len(img_file_path) = 0), then convert
+
+            for nifti_folder in nifti_folders:
+                #  Get the json files
+                base_file_name = os.path.basename(nifti_folder).split(".")[0]
+                json_file_paths = glob(
+                    path.join(subj_folder, "**/" + base_file_name + ".json"),
+                    recursive=True
+                )
+
+                # Looking for an existing func or anat folder
+                dir_name = path.dirname(nifti_folder)
+                regex = re.compile("(?P<folder>anat|func)[1-9]")
+                if regex.search(dir_name) is not None:
+                    output_folder = regex.search(dir_name).group("folder")
+                elif regex.search(path.abspath(dir_name)) is not None:
+                    output_folder = regex.search(
+                        path.abspath(dir_name)
+                    ).group("folder")
```

Comment:
Not sure I understand what's happening here. Can't you just filter with `if "anat" in path or "func" in path` according to your comment?

Reply:
I don't think so or I don't understand either :) I have to parse the path of the json or nifti, but the format of the OASIS3 files and folders is quite variable. For example, I can have : **/(anat|func)?/.*\.(json|nii\.gz). Either, the json or the nifti can be or not be in a folder called anat or func and be in a variable number of parent folders (1 or 2).

Reply:
Ok then, please add a comment prior to this code with your reasoning 👍